prime-dotnet-webapi/Models/Api/PaginatedList.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Prime.Models.Api
+{
+    public class PaginatedResponse<T>
+    {
+        public List<T> Results { get; set; }
+        public int Page { get; set; }
+        public int TotalPages { get; set; }
+        public int Total { get; set; }
+        public int PageSize { get; set; }
+
+        public PaginatedResponse(PaginatedList<T> list)
+        {
+            Results = list;
+            Page = list.Page;
+            TotalPages = list.TotalPages;
+            Total = list.Total;
+            PageSize = list.PageSize;
+        }
+    }
+    public class PaginatedList<T> : List<T>
+    {
+        private static int GetPageSize()
+        {
+            return 100;
+        }
+
+        public int Page { get; private set; }
+        public int TotalPages { get; private set; }
+        public int Total { get; private set; }
+        public int PageSize { get; private set; }
+
+        public PaginatedList(List<T> items, int count, int page, int pageSize)
```

Comment:
I would prefer the input parameter `items` to be as most generic as possible. e.g. `IEnumerable<T>` to indicate it's a container without the needs to add/remove elements.